bgpd/bgp_vty.c

```diff
@@ -1353,21 +1420,21 @@ DEFPY (no_bgp_router_id,
        "Override configured router identifier\n"
        "Manually configured router identifier\n")
 {
-	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	nb_cli_enqueue_change(vty, "./router-id", NB_OP_DESTROY,
+			      router_id_str ? router_id_str : NULL);
 
-	if (router_id_str) {
-		if (!IPV4_ADDR_SAME(&bgp->router_id_static, &router_id)) {
-			vty_out(vty, "%% BGP router-id doesn't match\n");
-			return CMD_WARNING_CONFIG_FAILED;
-		}
-	}
+	return nb_cli_apply_changes(vty, NULL);
+}
 
-	router_id.s_addr = 0;
-	bgp_router_id_static_set(bgp, router_id);
+void cli_show_router_bgp_router_id(struct vty *vty, struct lyd_node *dnode,
+				   bool show_defaults)
+{
+	struct in_addr router_id;
 
-	return CMD_SUCCESS;
-}
+	yang_dnode_get_ipv4(&router_id, dnode, NULL);
```

Comment:
nit: just use `yang_dnode_get_string()` instead to avoid converting the router-id to an IPv4 address and then back to a string.

As a rule of thumb, always prefer using `yang_dnode_get_string()` in the `cli_show()` callbacks (as suggested by the NB retrofitting documentation).